### PR TITLE
docs(main): add telemetry discussion to debug.md

### PR DESCRIPTION
**Pull request by quartzmo**
refs: googleapis/google-api-go-client#2127
refs: #8655

### CODE REVIEW

**Review by bhshkh — APPROVED** [review]

**Review by codyoss — APPROVED** [review]

**Review by hongalex — APPROVED** [review]

**Review by tritone — COMMENTED** [review]
A few thoughts; we discussed offline but let me know if you have further questions.

File: debug.md [hunk 1]
```diff
@@ -160,3 +170,147 @@ func main() {
...
+[OpenTelemetry](https://opentelemetry.io/docs/what-is-opentelemetry/) is now
+available. The transition from OpenCensus to OpenTelemetry is covered in the
+following sections.
+
+### Tracing (experimental)
+
+All clients contain experimental support for the propagation of both OpenCensus
+and OpenTelemetry trace context to their receiving endpoints. However, apart
+from spans created during context propagation, the generated clients do not
+create spans. Only the following hand-written clients create spans for select
```
[the remark this answers is not in the document]
Comment (quartzmo) [reply debug.md]: Updated to (hopefully) clarify that the only spans in scope for this discussion (and affected by the changes discussed) are those in handwritten clients created via the `internal/trace` wrapper. Let me know if you think I should take another pass at it. Suggestions welcome.

File: debug.md [hunk 2]
```diff
@@ -160,3 +170,147 @@ func main() {
...
+        option.WithScopes(storage.ScopeFullControl),
+)
+if err != nil {
+        log.Fatal(err)
+}
+// An example of passing options to the otelhttp.Transport.
+otelOpts := otelhttp.WithFilter(func(r *http.Request) bool {
+	return r.URL.Path != "/ping"
+})
+hc := &http.Client{
```
[the remark this answers is not in the document]
Comment (quartzmo) [reply debug.md:344]: Good idea!

File: debug.md [hunk 3]
```diff
@@ -160,3 +170,147 @@ func main() {
...
+OpenTelemetry, the successor project.
+
+Using the [OpenTelemetry-Go - OpenCensus Bridge](https://github.com/open-telemetry/opentelemetry-go/tree/main/bridge/opencensus), you can immediately begin exporting your traces with OpenTelemetry, even while
+dependencies of your application remain instrumented with OpenCensus. If you do
+not use the bridge, you will need to migrate your entire application and all of
+its instrumented dependencies at once.  For simple applications, this may be
+possible, but we expect the bridge to be helpful if multiple libraries with
+instrumentation are used.
+
+On May 29, 2024, six months after the
```
[the remark this answers is not in the document]
Comment (quartzmo) [reply debug.md:214]: #8655 was released in 0.111.0, as linked in the line above.

File: debug.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (quartzmo) [reply debug.md:344]: Hmm, I already have the comment line above: `// An example of passing options to the otelhttp.Transport.`, isn't that enough?

[the remark this answers is not in the document]
Comment (quartzmo) [reply debug.md:344]: Should I change the wording of that comment?

**Pull request merged**